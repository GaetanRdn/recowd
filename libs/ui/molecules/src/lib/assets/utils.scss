@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
