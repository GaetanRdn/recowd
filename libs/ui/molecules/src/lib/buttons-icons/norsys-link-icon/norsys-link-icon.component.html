<a href="https://www.norsys.fr/" rcButtonIcon target="_blank">
  <rc-norsys-custom-icon></rc-norsys-custom-icon>
</a>
