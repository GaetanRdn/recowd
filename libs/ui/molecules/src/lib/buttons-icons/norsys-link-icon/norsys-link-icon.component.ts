import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ButtonDirective } from '@recowd/ui-atoms';
import { NorsysCustomIconComponent } from '../../custom-icons/norsys-custom-icon/norsys-custom-icon.component';

@Component({
  selector: 'rc-norsys-link-icon',
  standalone: true,
  imports: [ButtonDirective, NorsysCustomIconComponent],
  templateUrl: './norsys-link-icon.component.html',
  styleUrls: ['./norsys-link-icon.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NorsysLinkIconComponent {}
