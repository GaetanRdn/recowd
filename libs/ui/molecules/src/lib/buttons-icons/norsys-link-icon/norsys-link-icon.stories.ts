import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { NorsysLinkIconComponent } from './norsys-link-icon.component';

export default {
  title: 'molecules/buttons-icons/norsys-link-icon',
  parameters: {
    jest: 'norsys-link-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [NorsysLinkIconComponent],
    }),
  ],
} as Meta<NorsysLinkIconComponent>;

const Template: Story<NorsysLinkIconComponent> = () => ({
  template: `<rc-norsys-link-icon></rc-norsys-link-icon>`,
});

export const Default = Template.bind({});
