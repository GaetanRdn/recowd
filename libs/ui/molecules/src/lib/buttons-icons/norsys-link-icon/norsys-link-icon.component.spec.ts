import { TestBed, waitForAsync } from '@angular/core/testing';

import { NorsysLinkIconComponent } from './norsys-link-icon.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('NorsysLinkIconComponent', () => {
  let templateLookup: TemplateLookup<NorsysLinkIconComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [NorsysLinkIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(NorsysLinkIconComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot());
});
