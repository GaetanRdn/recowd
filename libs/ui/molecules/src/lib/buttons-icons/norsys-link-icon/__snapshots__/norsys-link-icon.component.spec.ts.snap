// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NorsysLinkIconComponent should create 1`] = `
HTMLCollection [
  <a
    class="rc-button-icon"
    href="https://www.norsys.fr/"
    rcbuttonicon=""
    target="_blank"
  >
    <rc-norsys-custom-icon>
      <rc-custom-icon>
        <object
          data="assets/images/norsys.svg"
          height="24"
          type="image/svg+xml"
          width="24"
        >
          <img
            alt="assets/images/norsys"
            src="assets/images/norsys.png"
          />
        </object>
      </rc-custom-icon>
    </rc-norsys-custom-icon>
  </a>,
]
`;
